packages/log-view/src/browser/log-view-widget.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { VirtualWidget, VirtualRenderer } from "@theia/core/lib/browser";
+import { LogModel, LogChannel, ALL_CHANNEL_NAME } from "./log-model";
+import { VirtualElement, h } from "@phosphor/virtualdom";
+
+import "../../src/browser/style/log-view.css";
+
+export const LOG_VIEW_KIND = 'logView';
+
+@injectable()
+export class LogViewWidget extends VirtualWidget {
+
+    private selectedChannel: LogChannel;
+
+    constructor( @inject(LogModel) protected readonly logModel: LogModel) {
+        super();
+        this.id = 'logView';
+        this.title.label = 'Log';
+        this.title.iconClass = 'fa fa-flag';
+        this.title.closable = true;
+        this.addClass('theia-log-view');
+        this.selectedChannel = logModel.getChannel(ALL_CHANNEL_NAME);
+        logModel.onContentChange(logChannel => {
+            if (logChannel === this.selectedChannel) {
+                this.update();
+            }
+        });
+        logModel.onChannelsChange(logChannels => {
+            this.update();
+        });
+        this.update();
+    }
+
+    render(): VirtualElement[] {
+        const contents = h.div({ id: 'logContents' }, this.selectedChannel.contents);
+        return [this.renderLogSelector(), contents];
+    }
+
+    private renderLogSelector(): VirtualElement {
+        const logChannelOptionElements: h.Child[] = [];
+        this.logModel.getChannelNames().forEach(channelName => {
+            logChannelOptionElements.push(h.option({ value: channelName }, channelName));
+        });
+
+        return h.select({
+            id: 'logChannelList',
+            onchange: async event => {
+                const channelName = (event.target as HTMLSelectElement).value;
+                this.selectedChannel = this.logModel.getChannel(channelName);
+                this.update();
+            }
+        }, VirtualRenderer.flatten(logChannelOptionElements));
+    }
+
+    update() {
```

Comment:
you should override `onUpdateRequest` instead

Reply:
Done.